lib/trino-hive-formats/src/main/java/io/trino/hive/formats/line/grok/Grok.java

```diff
@@ -205,13 +224,14 @@ private static void loadDefaultDatePatterns(String name, String conf)
         if (conf == null || conf.isBlank()) {
             throw new GrokException("Invalid value when loading config file");
         }
-        if (DEFAULT_DATE_FORMATS.containsKey(name)) {
-            DEFAULT_DATE_FORMATS.get(name).add(conf);
+
+        if (defaultConfig.containsKey(name)) {
+            ArrayList<String> existingFormats = new ArrayList<>(defaultConfig.get(name));
+            existingFormats.add(conf);
+            defaultConfig.put(name, ImmutableList.copyOf(existingFormats));
```

Comment:
No need to copy into and out of `ImmutableList` on every modification. You can use `Map<String, ImmutableList.Builer<String>>` while inserting values and then copy the result into the final immutable form with something like:
```
DEFAULT_DATE_FORMATS = ImmutableMap.copyOf(Maps.transform(temporaryMap, ImmutableList.Builder::build)));
```